beacon/blockchain/deposit.go

```diff
@@ -49,10 +51,48 @@ func (s *Service[
 	s.fetchAndStoreDeposits(ctx, blockNum-s.eth1FollowDistance)
 }
 
-// depositCatchupFetcher fetches deposits for blocks that failed to be
-// processed.
 func (s *Service[
-	_, _, _, _, _,
+	_, _, ConsensusBlockT, _, _, _, _, _, _, _, _, _, _, _,
+]) fetchAndStoreDeposits(
```

Comment:
correct me if I'm wrong. The two triggers for `fetchAndStoreDeposits` are:
1. `ProcessBeaconBlock` -> `depositFetcher`
2. `go depositCatchupFetcher` on a retry interval

nit: personally feels cleaner to me to have a "retrySet" object (that collects and serves the failed blocks) that handles concurrency itself, so we don't see all the locking/unlocking code in the main control flow. Not biased either way though.